app/code/Magento/Inventory/etc/di.xml

```diff
@@ -86,4 +86,10 @@
             <argument name="stockSourceLinkValidator" xsi:type="object">Magento\Inventory\Model\StockSourceLink\Validator\ValidatorChain</argument>
         </arguments>
     </type>
+    <type name="Magento\InventoryApi\Api\StockSourceLinksSaveInterface">
```

Comment:
We already have validation in 
`\Magento\InventoryCatalog\Model\StockSourceLink\Validator\AssignDefaultSourceToNonDefaultStockValidator`
need to change logic of current implamantation, but not create one more point

Reply:
@naydav, I've removed plugin and reworked validation.